inbc-program/tests/unit/test_source_app_parser.py

```diff
@@ -24,8 +26,76 @@ def test_parse_add_arguments_successfully(self):
+                                     'Components: main'])
+        self.assertEqual(f.filename, 'intel-gpu-jammy.list')
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_raise_application_add_with_only_one_gpgKeyUri_param(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--gpgKeyUri', 'https://repositories.intel.com/gpu/intel-graphics.key',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        with pytest.raises(InbcException,
+                           match="Source requires either both gpgKeyUri and gpgKeyName "
+                                 "to be provided, or neither of them."):
+            Inbc(p, 'source', False)
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_raise_application_add_with_only_one_gpgKeyName_param(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--gpgKeyName', 'intel-graphics.gpg',
+             '--sources', 'deb http://example.com/ focal main restricted universe',
+             'deb-src http://example.com/ focal-security main',
+             '--filename', 'intel-gpu-jammy.list'])
+        with pytest.raises(InbcException,
+                           match="Source requires either both gpgKeyUri and gpgKeyName "
+                                 "to be provided, or neither of them."):
+            Inbc(p, 'source', False)
+
+    @patch('inbm_lib.mqttclient.mqtt.mqtt.Client.connect')
+    def test_create_add_deb_822_format_manifest_successfully(self, m_connect):
+        p = self.arg_parser.parse_args(
+            ['source', 'application', 'add',
+             '--sources', 'X-Repolib-Name: Google Chrome',
+                          'Enabled: yes',
+                          'Types: deb',
+                          'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
+                          'Suites: stable',
+                          'Components: main',
+             '--filename', 'google-chrome.sources'])
+        Inbc(p, 'source', False)
+        expected = '<?xml version="1.0" encoding="utf-8"?><manifest><type>source</type><applicationSource>' \
+                   '<add><repo><repos><source_pkg>X-Repolib-Name: Google Chrome</source_pkg>' \
+                   '<source_pkg>Enabled: yes</source_pkg>' \
+                   '<source_pkg>Types: deb</source_pkg>'\
+                   '<source_pkg>URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub</source_pkg>' \
+                   '<source_pkg>Suites: stable</source_pkg>' \
+                   '<source_pkg>Components: main</source_pkg>' \
+                   '</repos><filename>google-chrome.sources</filename></repo></add></applicationSource></manifest>'
```

Comment:
**Suggestion:** Refactor the repeated XML manifest string construction into a helper method to avoid duplication and improve code readability. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        def construct_manifest(source_packages, filename):
            source_pkg_elements = ''.join(f'<source_pkg>{pkg}</source_pkg>' for pkg in source_packages)
            return f'<?xml version="1.0" encoding="utf-8"?><manifest><type>source</type><applicationSource>' \
                   f'<add><repo><repos>{source_pkg_elements}</repos><filename>{filename}</filename></repo></add></applicationSource></manifest>'

        source_packages = [
            'X-Repolib-Name: Google Chrome',
            'Enabled: yes',
            'Types: deb',
            'URIs: https://dl-ssl.google.com/linux/linux_signing_key.pub',
            'Suites: stable',
            'Components: main'
        ]
        expected = construct_manifest(source_packages, 'google-chrome.sources')
```